internal/kafka/internal/handlers/observatorium_proxy.go

```diff
@@ -0,0 +1,58 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/auth"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/handlers"
+	"github.com/gorilla/mux"
+)
+
+type observatoriumProxyHandler struct {
+	clusterService services.ClusterService
+}
+
+func NewObservatoriumProxyHandler(clusterService services.ClusterService) *observatoriumProxyHandler {
+	return &observatoriumProxyHandler{
+		clusterService: clusterService,
+	}
+}
+
+// ValidateTokenAndExternalClusterID validates combination of external cluster ID parameter against client ID from the claims
+func (h observatoriumProxyHandler) ValidateTokenAndExternalClusterID(w http.ResponseWriter, r *http.Request) {
+	external_id := mux.Vars(r)["cluster_external_id"]
```

Comment:
Is there some reason the ID provided is ExternalID instead of ClusterID? for cluster agent endpoints ClusterID is used

Reply:
It's because the `cluster_id` label metric actually represents the `cluster.ExternalID` (which is the physical cluster spec ID) in KFM database and not the `cluster.ClusterID` which is an OCM id (primary key)

Reply:
I understand your comment, but how is the value being set for the metric label `cluster_id related to what the data plane sends to the kafka request in the URL? They are independent.

Reply:
> but how is the value being set for the metric label `cluster_id related to what the data plane sends to the kafka request in the URL? 

I am not sure I understood this part. But I'll try to respond based on what I've understood from the question.
First of all, this a Kafka metric's only endpoint. It has nothing to do with what the fleetshard sends to us via the agents-endpoints.

This endpoint is called by the observatorium-proxy to verify if a given clientID (in the token) is authorized for a given cluster external id before metrics are forwarded/proxied to observatorium.
The proxy gets the metric (and not the Kafka request) data from Prometheus remote write - this has an override to ensure that the cluster_id label contains the value i.e the external_id.

This id being the cluster uuid (ocm's external_id) it seems consistency for the URL parameter to speak as such to avoid such confusions e.g
The URL parameter is called `cluster_id` but why are we comparing it against `cluster.ExteranalID` and not `cluster.ClusterID` 

Note: The `cluster_id` that's seen in Agents endpoints is a `cluster.ClusterID` as represented in the `Cluster` entity.

Reply:
Can the observatorium proxy send the ClusterID instead of ExternalID?

Reply:
@miguelsorianod the proxy gets the clusterId from the label on the samples in the remote write request.

Reply:
I think the question is, can the observability-operator configured to send the ClusterID, instead of fetching ExternalID from the cluster ? I'll leave that to @pb82 to chime in for more details but as per other previously discussions (precisely about this value) this value is used somewhere - to construct the dashboards and/or  sub watch (I can't remember exactly where in dashboards). An additional question, is the change worth it?

Reply:
> @miguelsorianod the proxy gets the clusterId from the label on the samples in the remote write request.

I understand you mean the External Cluster ID.
Is there another label available that contains the Cluster ID? If yes, could that be used alternatively to perform the request? If not, would it make sense to add it?